### PR TITLE
[fix] searx.network: fix rare cases where LOOP is None

### PR DESCRIPTION
**Pull request by dalf**
## What does this PR do?

* searx.network.client.LOOP is initialized in a thread
* searx.network.__init__ imports LOOP which may happen
  before the thread has initialized LOOP

This commit adds a new function "searx.network.client.get_loop()"
to fix this issue

## Why is this change important?

With uwsgi, some worker may be broken when LOOP is None

## How to test this PR locally?

Not easily reproducible.

### CODE REVIEW

File: searx/network/client.py [hunk 1]
```diff
@@ -136,7 +135,7 @@ def get_transport_for_socks_proxy(verify, http2, local_address, proxy_url, limit
     return AsyncProxyTransportFixed(proxy_type=proxy_type, proxy_host=proxy_host, proxy_port=proxy_port,
                                     username=proxy_username, password=proxy_password,
                                     rdns=rdns,
-                                    loop=LOOP,
+                                    loop=get_loop(),
```
Comment (return42) [inline searx/network/client.py:138]: I rebased and pushed force ... BTW I added this replacement for the global LOOP variable .. I guess you missed it.

File: searx/network/client.py [hunk 2]
```diff
@@ -192,6 +191,11 @@ def new_client(enable_http, verify, enable_http2,
     return httpx.AsyncClient(transport=transport, mounts=mounts, max_redirects=max_redirects)
 
 
+def get_loop():
+    global LOOP
```
Comment (dalf) [inline searx/network/client.py:195]: There is no need `global LOOP` since LOOP is not assigned (?)

File: searx/network/client.py (hunk 1 above)
Comment (dalf) [reply searx/network/client.py:138]: Can you add another commit rather than push force, so I can easy see what has been changed ?

Comment (return42) [reply searx/network/client.py:138]: > Can you add another commit

sure .. sorry.

Comment (dalf) [reply searx/network/client.py:138]: NP

File: searx/network/client.py (hunk 2 above)
Comment (return42) [reply searx/network/client.py:195]: > There is no need global LOOP since LOOP is not assigned (?)

To access a global variable inside a function there is no need to use global keyword in general .. but I think it is good practice to use global keyword to explicit define a variable as _global_ instead of being [implicit](https://docs.python.org/3/faq/programming.html#what-are-the-rules-for-local-and-global-variables-in-python).

BTW: LOOP is assigned by `None` in line 25  / otherwise you would get an `NameError: name 'LOOP' is not defined`:

https://github.com/searxng/searxng/blob/283ae7bfad05fedbf6c7a7e7d7addfc3ae7ed9c7/searx/network/client.py#L25

TLDR: compared to C++ in real object oriented languages types are also objects, here is the id of the `None` type object:

    >>> id(None)
    9480720

Comment (dalf) [reply searx/network/client.py:195]: >  I think it is good practice to use global keyword to explicit define a variable as global instead of being implicit. 

Ok.
But new code will have a lot of `global` since there are a lot of global variables.

Comment (return42) [reply searx/network/client.py:195]: [On one hand, requiring global for assigned variables provides a bar against unintended side-effects. On the other hand, if global was required for all global references, you’d be using global all the time. You’d have to declare as global every reference to a built-in function or to a component of an imported module. This clutter would defeat the usefulness of the global declaration for identifying side-effects.](https://docs.python.org/3/faq/programming.html#what-are-the-rules-for-local-and-global-variables-in-python)

> But new code will have a lot of global since there are a lot of global variables.

Hm ... if it is to much, then we can switch to use implicit globals .. lets see ..I can't judge which is better for searx / I do use globals very rare in my projects.

**Comment by return42** [comment]
If you think this PR is ready to merge .. please merge.  I can't give a _approved_ from github-GUI / I guess it is because I'm in the list of _committers_ of this PR.
